### ISSUE TITLE
compare_columns not excluding columns since > 0.2.1 on BigQuery

### ISSUE THREAD

**Issue by turner-atg** [mjirv/dbt-datamocktool#66]
### Describe the bug
When using datamocktools >= 0.2.2 with `dbt-bigquery==1.3.1  and `dbt-core==1.3.2`, `dbt compile` reports an error as a result of attempting to compare columns with unsupported types.

### Steps to reproduce
1. Create test inputs with ARRAY types.
2. Create test output.
3. Select compare_columns to not include the ARRAY type columns.
4. run `dbt compile`

### Expected results
dbt compile should not raise an error.

### Actual results
```
19:02:43  BigQuery adapter: ***
19:02:43  Encountered an error:
Runtime Error
  Database Error in test test_census_aggregations (models/marts/my_model/_models.yml)
    Column 1 in INTERSECT DISTINCT has type that does not support set operation comparisons: ARRAY at [341:5]
```

Clicking on the link provided by the bigquery adapter shows that *all* columns are being compared. 

### System information
**The contents of your `packages.yml` file:**
```
packages:
  - package: dbt-labs/dbt_utils
    version: 1.0.0
  - package: mjirv/dbt_datamocktool
    version: 0.2.2
```

**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [ x] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
Core:
  - installed: 1.3.2
  - latest:    1.4.3 - Update available!

  Your version of dbt-core is out of date!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

Plugins:
  - bigquery: 1.3.1 - Update available!

  At least one plugin is out of date or incompatible with dbt-core.
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

```

### Additional context

Looks like it might be related to the switch to [`audit_helper.compare_relations`](https://github.com/mjirv/dbt-datamocktool/blob/ad0619e5c9ad54aaa2e1fe8f87c768b3e39581d7/macros/dmt_unit_test.sql#L41). Maybe just a bug with the way the exclude_columns list is being generated. Not sure why it's being mapped to upper.

### Are you interested in contributing the fix?

Happy to contribute the fix. Pretty experienced with python/SQL, but not familiar with the dbt python API or bigquery adapters. Would need help getting started.

**Comment by millyleadley** [comment]
I ran into the same thing, [this PR](https://github.com/mjirv/dbt-datamocktool/pull/67) solved it for me.

Separately, I'd quite like to implement an optional `exclude_columns` argument which means it'll test all the columns apart from the ones stated in a list. Like you, I have an array column which doesn't play nicely with `INTERSECT DISTINCT` and I'd like to exclude it

**Comment by mjirv** [comment]
Fixed by #67
